x-pack/plugins/viz_editor/indexpattern_datasource_plugin/indexpattern_datasource_plugin.tsx

```diff
@@ -4,47 +4,75 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-// @ts-ignore
-import { EuiSuperSelect } from '@elastic/eui';
-import zipObject from 'lodash-es/zipObject';
+import { EuiComboBox } from '@elastic/eui';
 import React, { useEffect, useState } from 'react';
-import { IndexPattern } from 'ui/index_patterns';
 import { FieldListPanel } from '../public/common/components/field_list_panel';
-import { VisModel } from '../public/common/lib';
+import { Datasource, VisModel } from '../public/common/lib';
 import { DatasourcePlugin, PanelComponentProps } from '../public/datasource_plugin_registry';
 import { getIndexPatterns } from './index_patterns';
 
+interface DataPanelState {
+  indexPatterns: Datasource[];
+  selectionTitle: string;
```

Comment:
Why do we need the selection title in the local state? Isn't the same information available in `visModel.datasource.title`?

Reply:
You're right, will update